### PR TITLE
Add property types and return types in constraints

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Forgot a "few" ones in constraints :smile:

### CODE REVIEW

File: doctrine/resolve_target_entity.rst [hunk 1]
```diff
@@ -69,7 +69,7 @@ An Invoice entity::
          * @var InvoiceSubjectInterface
```
Comment (OskarStark) [inline doctrine/resolve_target_entity.rst]: Can be removed then

File: reference/constraints/Collection.rst [hunk 2]
```diff
@@ -261,9 +263,9 @@ you can do the following:
 
         class Author
         {
-            protected $profileData = ['personal_email'];
```
Comment (OskarStark) [inline reference/constraints/Collection.rst:264]: ?

File: reference/constraints/GreaterThan.rst [hunk 3]
```diff
@@ -112,7 +114,7 @@ that a date must at least be the next day:
         class Order
         {
             #[Assert\GreaterThan('today')]
-            protected $deliveryDate;
+            protected \DateTime $deliveryDate;
```
Comment (OskarStark) [inline reference/constraints/GreaterThan.rst]:
```suggestion
            protected \DateTimeInterface $deliveryDate;
```
?

File: reference/constraints/GreaterThan.rst [hunk 4]
```diff
@@ -169,7 +173,7 @@ dates. If you want to fix the timezone, append it to the date string:
         class Order
         {
             #[Assert\GreaterThan('today UTC')]
-            protected $deliveryDate;
+            protected \DateTime $deliveryDate;
```
Comment (OskarStark) [inline reference/constraints/GreaterThan.rst]: same for the others?

File: reference/constraints/Collection.rst (hunk 2 above)
Comment (alexandre-daubois) [reply reference/constraints/Collection.rst:264]: I think it was wrong at first, it should have been `protected array $profileData = ['personal_email' => 'email@example.com'];`, like in the attributes example. But I saw that plenty examples of `loadValidatorMetadata` used ellipsis in place of properties declarations. So I removed it.

Should it be reverted and updated to `protected array $profileData = ['personal_email' => 'email@example.com'];` ?

Comment (OskarStark) [reply reference/constraints/Collection.rst:264]: Ok lets remove it

File: reference/constraints/GreaterThan.rst (hunk 3 above)
Comment (alexandre-daubois) [reply reference/constraints/GreaterThan.rst]: Updated!

**Comment by javiereguiluz** [comment]
Alex, thanks a million for the colossal effort of adding so many types 🙇🙇🙇

**Pull request merged**